awx/api/serializers.py

```diff
@@ -1533,7 +1533,8 @@ def get_related(self, obj):
 class InventorySourceOptionsSerializer(BaseSerializer):
 
     class Meta:
-        fields = ('*', 'source', 'source_path', 'source_script', 'source_vars', 'credential',
+        fields = ('*', 'source', 'source_path', 'source_script', 'source_vars',
+                  'credential', 'vault_credential',
```

Comment:
Actually, this will need to copy the validation used in job templates.

Reply:
👍

Reply:
Added that, but by way of the `clean_vault_credential` on model.